staging/cse/windows/provisioningscripts/kubeproxystart.ps1

```diff
@@ -50,11 +51,15 @@ if ($Global:ClusterConfiguration.Kubernetes.Kubeproxy.ConfigArgs) {
     $global:KubeproxyArgList += $Global:ClusterConfiguration.Kubernetes.Kubeproxy.ConfigArgs
 }
 
-$hnsNetwork = Get-HnsNetwork | ? Name -EQ $KubeNetwork
-while (!$hnsNetwork) {
-    Write-Host "$(Get-Date -Format o) Waiting for Network [$KubeNetwork] to be created . . ."
-    Start-Sleep 10
+if ($IsCleanupHnsNetwork) { # CleanupHnsNetwork is legacy code
     $hnsNetwork = Get-HnsNetwork | ? Name -EQ $KubeNetwork
+    while (!$hnsNetwork) {
+        Write-Host "$(Get-Date -Format o) Waiting for Network [$KubeNetwork] to be created . . ."
+        Start-Sleep 10
+        $hnsNetwork = Get-HnsNetwork | ? Name -EQ $KubeNetwork
```

Comment:
Assuming IsCleanupHnsNetwork set to false, this code will never run, right? We should not be removing HNS network on kube-proxy restart. Kubeproxy only manages policylists. Kube-proxy will delete policylists itself when it is started, so I don't see any reason to delete any HNS resources. It should never be run imho.